libraries/load-themed-styles/src/index.ts

```diff
@@ -83,12 +151,44 @@ export function loadStyles(styles: string | ThemableArray): void {
  * a loadStyles callback that gets called when styles are loaded or reloaded
  */
 export function configureLoadStyles(
-    loadStyles: ((processedStyles: string, rawStyles?: string | ThemableArray) => void) | undefined
-  ): void {
+  loadStyles: ((processedStyles: string, rawStyles?: string | ThemableArray) => void) | undefined
+): void {
   _themeState.loadStyles = loadStyles;
 }
 
 /**
+ * Configure run mode of load-themable-styles
+ * @param mode load-themable-styles run mode, async or sync
+ */
+export function configureRunMode(mode: Mode): void {
+  _themeState.runState.mode = mode;
+}
+
+/**
+ * external code can call flush to synchronously force processing of currently buffered styles
+ */
+export function flush(): void {
+  measure(() => {
+    const styleArrays: ThemableArray[] = _themeState.runState.buffer.slice();
+    _themeState.runState.buffer = [];
+    const mergedStyleArray: ThemableArray[] = [].concat.apply([], styleArrays);
+    if (mergedStyleArray.length > 0) {
+      applyThemableStyles(mergedStyleArray);
+    }
+  });
+}
+
+/**
+ * register async loadStyles
+ */
+function asyncLoadStyles(): number {
+  return setTimeout(() => {
+    _themeState.runState.flushTimer = 0;
+    flush();
+  }, 0);
```

Comment:
setTimeout 0 means browser finishes the current sync work stack and then work on the given function immediately. All bundle code parsing and  styles loading are sync work. It will finish style parsing and loading to buffer all css in that bundle and then flush them..  Then next bundle...